src/smt/preprocess_proof_generator.cpp

```diff
@@ -21,9 +21,13 @@
 namespace CVC4 {
 namespace smt {
 
-PreprocessProofGenerator::PreprocessProofGenerator(context::UserContext* u,
-                                                   ProofNodeManager* pnm)
-    : d_pnm(pnm), d_src(u), d_helperProofs(u)
+PreprocessProofGenerator::PreprocessProofGenerator(ProofNodeManager* pnm,
+                                                   context::Context* c,
+                                                   std::string name)
+    : d_pnm(pnm),
+      d_src(c ? c : &d_context),
```

Comment:
I'm confused about where `d_context` is set?

Reply:
d_context is a non-pointer data member with a default constructor, one could put `d_context()` above but its not necessary.

Reply:
I see. I forgot how this worked. :)